hardhat.config.ts

```diff
@@ -156,7 +164,7 @@ const config: HardhatUserConfig = {
     timeout: 10000 * 1000,
   },
   etherscan: {
-    apiKey: String(process.env.SCANAPI),
+    apiKey: getScanApiKey(String(process.env.networkType)),
```

Comment:
Let's update the hardhat etherscan version as well.

Reply:
Sure